File: Linguagem de Programação I/Extras/Beecrowd/1023.py
import sys
import math

T = 0
for line in sys.stdin:
    N = int(line)
    if N == 0:
        break

    if T:
        print('')

    totalX, totalY = 0, 0
    consumos = {}
    for _ in range(N):
        X, Y = map(int, sys.stdin.readline().split())
        totalX += X
        totalY += Y
        consumos[Y // X] = consumos.get(Y // X, 0) + X

    T += 1
    print(f'Cidade# {T}:')

    output = []
    keys = sorted(consumos.keys())
    for key in keys:
        output.append(f'{consumos[key]}-{key}')

    print(' '.join(output))

    consumo_medio = math.floor((100 * totalY) / totalX) / 100
    print(f'Consumo medio: {consumo_medio:.2f} m3.')

File: Linguagem de Programação I/Extras/Beecrowd_Extra/1023.rs
use std::io::{self, BufRead};
use std::collections::BTreeMap;

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();

    let mut T = 0;
    loop {
        let line = lines.next().unwrap().unwrap();
        let N: i32 = line.trim().parse().unwrap();
        if N == 0 {
            break;
        }

        if T > 0 {
            println!();
        }

        let mut totalX = 0;
        let mut totalY = 0;
        let mut consumos = BTreeMap::new();
        for _ in 0..N {
            let line = lines.next().unwrap().unwrap();
            let inputs: Vec<i32> = line.split_whitespace().map(|x| x.parse().unwrap()).collect();
            let (X, Y) = (inputs[0], inputs[1]);
            totalX += X;
            totalY += Y;
            let key = Y / X;
            *consumos.entry(key).or_insert(0) += X;
        }

        T += 1;
        println!("Cidade# {}:", T);

        let mut output = Vec::new();
        for (key, value) in &consumos {
            output.push(format!("{}-{}", value, key));
        }

        println!("{}", output.join(" "));

        let consumo_medio = ((100.0 * totalY as f64) / totalX as f64).floor() / 100.0;
        println!("Consumo medio: {:.2} m3.", consumo_medio);
    }
}